src/lib/plugin.js

```diff
@@ -29,73 +29,107 @@ module.exports = class PluginVirtual extends EventEmitter2 {
   constructor (opts) {
     super()
 
-    assertOptionType(opts, 'currencyCode', 'string')
-    assertOptionType(opts, 'currencyScale', 'number')
-    assertOptionType(opts, 'maxBalance', 'string')
-    assertOptionType(opts, 'secret', 'string')
-    assertOptionType(opts, 'peerPublicKey', 'string')
-    assertOptionType(opts, '_store', 'object')
-    assertOptionType(opts, 'rpcUri', 'string')
-
-    this._secret = opts.secret
-    this._peerPublicKey = opts.peerPublicKey
-    this._publicKey = Token.publicKey(this._secret)
-    this._currencyCode = opts.currencyCode.toUpperCase()
-    this._currencyScale = opts.currencyScale
+    this._stateful = !!opts._store
+    if (this._stateful) {
+      assertOptionType(opts, 'currencyCode', 'string')
+      assertOptionType(opts, 'currencyScale', 'number')
+      assertOptionType(opts, 'maxBalance', 'string')
+
+      this._store = opts._store
+      this._maxBalance = opts.maxBalance
+      this._minBalance = opts.minBalance
+      this._highestBalance = new Balance({
+        key: 'balance__',
+        store: this._store,
+        maximum: this._maxBalance
+      })
+
+      this._lowestBalance = new Balance({
+        key: 'balance_l',
+        minimum: this._minBalance,
+        store: this._store
+      })
+
+      // give a 'balance' event on balance change
+      this._highestBalance.on('balance', (balance) => {
+        this.emit('balance', balance)
+      })
+    }
 
-    this._store = opts._store
-    this._maxBalance = opts.maxBalance
-    this._balance = new Balance({
-      maximum: this._maxBalance,
-      store: this._store
-    })
+    if (opts.rpcUris) {
```

Comment:
I don't think we gain anything from putting it in the standard, because it's configured per-plugin already